redaxo/src/addons/media_manager/lib/managed_media.php

```diff
@@ -165,46 +165,47 @@ public function setFormat($format)
 
     public function sendMedia($sourceCacheFilename, $headerCacheFilename, $save = false)
     {
-        $src = $this->getSource();
-
-        $this->prepareHeaders($src);
+        $this->prepareHeaders();
 
-        rex_response::cleanOutputBuffers();
-        foreach ($this->header as $t => $c) {
-            header($t . ': ' . $c);
-        }
+        if ($this->asImage) {
+            $src = $this->getSource();
+            $this->setHeader('Content-Length', rex_string::size($src));
 
-        $outputted = false;
-        // dependency ramsey/http-range requires PHP >=5.6
-        if (PHP_VERSION_ID >= 50600) {
-            header('Accept-Ranges: bytes');
-            $rangeHeader = rex_request::server('HTTP_RANGE', 'string', null);
-            if ($rangeHeader) {
-                try {
-                    $mediaSize = rex_string::size($src);
-
-                    $unitFactory = new \Ramsey\Http\Range\UnitFactory();
-                    $ranges = $unitFactory->getUnit(trim($rangeHeader), $mediaSize)->getRanges();
-                    foreach ($ranges as $range) {
-                        header('HTTP/1.1 ' . rex_response::HTTP_PARTIAL_CONTENT);
-                        header('Content-Length: ' . $range->getLength());
-                        header('Content-Range: bytes ' . $range->getStart() . '-' . $range->getEnd() . '/' . $mediaSize);
-
-                        echo substr($src, $range->getStart(), $range->getLength());
-                        $outputted = true;
-                    }
-                } catch (\Ramsey\Http\Range\Exception\HttpRangeException $exception) {
-                    header('HTTP/1.1 ' . rex_response::HTTP_RANGE_NOT_SATISFIABLE);
-                    $outputted = true;
-                }
+            rex_response::cleanOutputBuffers();
+            foreach ($this->header as $t => $c) {
+                header($t . ': ' . $c);
             }
-        }
-        if (!$outputted) {
+
             echo $src;
-        }
 
-        if ($save) {
-            $this->saveFiles($src, $sourceCacheFilename, $headerCacheFilename);
+            if ($save) {
+                rex_file::putCache($headerCacheFilename, [
+                    'media_path' => $this->getMediaPath(),
+                    'format' => $this->format,
+                    'headers' => $this->header,
+                ]);
+
+                rex_file::put($sourceCacheFilename, $src);
+            }
+        } else {
+            $this->setHeader('Content-Length', filesize($this->getSourcePath()));
```

Comment:
wie gelangt dieser header in die ausgabe? ich sehe nur das set-aber nirgends das echo

Reply:
durch Zeile 194 `foreach ($this->header as $t => $c) {`